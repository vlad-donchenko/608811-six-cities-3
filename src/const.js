const ZOOM = 12;

const MAX_NEIGHBOURHOOD_OFFERS = 3;

const monthMap = {
  "0": `January`,
  "1": `February`,
  "2": `March`,
  "3": `April`,
  "4": `May`,
  "5": `June`,
  "6": `July`,
  "7": `August`,
  "8": `September`,
  "9": `October`,
  "10": `November`,
  "11": `December`,
};

const OfferPrefix = {
  MAIN: `cities__place-card`,
  DETAILS: `near-places__card`,
};

export {ZOOM, MAX_NEIGHBOURHOOD_OFFERS, OfferPrefix, monthMap};
