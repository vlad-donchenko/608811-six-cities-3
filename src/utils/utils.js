const monthMap = {
  "0": `January`,
  "1": `February`,
  "2": `March`,
  "3": `April`,
  "4": `May`,
  "5": `June`,
  "6": `July`,
  "7": `August`,
  "8": `September`,
  "9": `October`,
  "10": `November`,
  "11": `December`,
};


const getRandomInteger = (min, max) => {
  const rand = min + Math.random() * (max + 1 - min);
  return Math.floor(rand);
};

const randomDate = (start, end) => {
  return new Date(start.getTime() + Math.random() * (end.getTime() - start.getTime()));
};

const formatDate = (date) => {
  const day = date.getDate();
  const year = date.getFullYear();
  const month = monthMap[date.getMonth()];

  return `${month} ${day} ${year}`;
};

const formatDateTime = (date) => {
  const day = date.getDate() < 10 ? `0${date.getDate()}` : `${date.getDate()}`;
  const year = date.getFullYear();
  const month = date.getMonth() < 10 ? `0${date.getMonth()}` : `${date.getMonth()}`;

  return `${year}-${month}-${day}`;
};

export {getRandomInteger, randomDate, formatDate, formatDateTime};
