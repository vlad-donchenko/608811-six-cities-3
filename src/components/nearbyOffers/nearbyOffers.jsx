import React from "react";
import OffersNearbyList from "../offers-nearby-list/offers-nearby-list.jsx";
import Map from "../map/map.jsx";
import {OfferPrefix} from "../../const";
import {offersType, onTitleClickType, onOfferMouseLeaveType, onOfferMouseEnterType, hoveredOfferType} from "../../types/index";

const NearbyOffers = ({offers, onTitleClick, onOfferMouseLeave, onOfferMouseEnter, hoveredOffer}) => {
  const additionalClass = OfferPrefix.NEAR_PLACES_CARD;

  return (
    <React.Fragment>
      <section className="property__map map">
        <Map offers={offers} hoveredOffer={hoveredOffer}/>
      </section>
      <div className="container">
        <OffersNearbyList offers={offers} onTitleClick={onTitleClick} onOfferMouseLeave={onOfferMouseLeave} onOfferMouseEnter={onOfferMouseEnter} additionalClass={additionalClass}/>
      </div>
    </React.Fragment>
  );
};

NearbyOffers.defaultProps = {
  hoveredOffer: undefined
};

NearbyOffers.propTypes = {
  offers: offersType,
  onTitleClick: onTitleClickType,
  onOfferMouseLeave: onOfferMouseLeaveType,
  onOfferMouseEnter: onOfferMouseEnterType,
  hoveredOffer: hoveredOfferType
};

export default NearbyOffers;
