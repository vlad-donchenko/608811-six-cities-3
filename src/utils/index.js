import {monthMap, sortType} from "../const";

const formatDate = (date) => {
  const dateFormat = new Date(date);
  const day = dateFormat.getDate();
  const year = dateFormat.getFullYear();
  const month = monthMap[dateFormat.getMonth()];

  return `${month} ${day} ${year}`;
};

const formatDateTime = (date) => {
  const dateFormat = new Date(date);
  const day = dateFormat.getDate() < 10 ? `0${dateFormat.getDate()}` : `${dateFormat.getDate()}`;
  const year = dateFormat.getFullYear();
  const month = dateFormat.getMonth() < 10 ? `0${dateFormat.getMonth()}` : `${dateFormat.getMonth()}`;

  return `${year}-${month}-${day}`;
};

const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const getOfferList = (offers, city) => {
  return offers.slice().filter((offer) => {
    return offer.city.name === city;
  });
};

const getLowToHighPriceOffers = (offers) => {
  return offers.slice().sort((a, b) => {
    return a.price - b.price;
  });
};

const getHighToLowPriceOffers = (offers) => {
  return offers.slice().sort((a, b) => {
    return  b.price - a.price;
  });
};

const getTopRatedOffers = (offers) => {
  return offers.slice().sort((a, b) => {
    return b.rating - a.rating;
  });
};

const sortingOffers = (activeSortType, offers) => {
  let sortedOffers = [];

  switch (activeSortType) {
    case sortType.popular :
      sortedOffers = offers;
      break;
    case sortType.toHigh :
      sortedOffers = getLowToHighPriceOffers(offers);
      break;
    case sortType.toLow :
      sortedOffers = getHighToLowPriceOffers(offers);
      break;
    case sortType.topRated :
      sortedOffers = getTopRatedOffers(offers);
  }

  return sortedOffers;
};

export {formatDate, formatDateTime, extend, getOfferList, sortingOffers};
