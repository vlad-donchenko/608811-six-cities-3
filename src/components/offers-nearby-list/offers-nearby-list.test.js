import React from "react";
import renderer from "react-test-renderer";
import OffersNearbyList from "./offers-nearby-list.jsx";
import nearbyOffers from "../../mocks/nearby-offers";
import {getOfferList} from "../../utils";
import {MAX_NEIGHBOURHOOD_OFFERS, CITIES, OfferPrefix} from "../../const";
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

const mockStore = configureStore([]);
const offers = getOfferList(nearbyOffers, CITIES[0]);

it(`Should render <OffersNearbyList/>`, () => {
  const store = mockStore({
    hoveredOfferId: null
  });

  const tree = renderer.create(<Provider store={store}><OffersNearbyList additionalClass={OfferPrefix.NEAR_PLACES_CARD} offers={offers.slice(0, MAX_NEIGHBOURHOOD_OFFERS)} onTitleClick={() => {
  }}/></Provider>).toJSON();

  expect(tree).toMatchSnapshot();
});
