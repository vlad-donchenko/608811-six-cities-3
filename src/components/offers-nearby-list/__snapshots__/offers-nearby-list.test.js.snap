// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <OffersNearbyList/> 1`] = `
<section
  className="near-places places"
>
  <h2
    className="near-places__title"
  >
    Other places in the neighbourhood
  </h2>
  <div
    className="near-places__list places__list"
  >
    <article
      className="near-places__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      
      <div
        className="near-places__card__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Sofitel Paris Le Faubourg - Nearby"
            className="place-card__image"
            height="200"
            src="img/apartment-02.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              1100
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button "
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            Sofitel Paris Le Faubourg - Nearby
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
    <article
      className="near-places__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="near-places__card__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Sofitel Paris Le Faubourg - Nearby"
            className="place-card__image"
            height="200"
            src="img/apartment-02.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              100
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button place-card__bookmark-button--active"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            Sofitel Paris Le Faubourg - Nearby
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
    <article
      className="near-places__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="near-places__card__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Sofitel Paris Le Faubourg - Nearby"
            className="place-card__image"
            height="200"
            src="img/apartment-03.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              190
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button "
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            Sofitel Paris Le Faubourg - Nearby
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
  </div>
</section>
`;
